src/components/course-listing.cjsx

```diff
@@ -0,0 +1,76 @@
+_ = require 'underscore'
+React = require 'react'
+BS = require 'react-bootstrap'
+Router = require 'react-router'
+
+{CourseListingActions, CourseListingStore} = require '../flux/course-listing'
+
+# Called once the store is loaded
+# checks the course and roles and will redirect if there is only a single course and role
+DisplayOrRedirect = (transition, callback) ->
+  courses = CourseListingStore.allCourses() or []
+  if courses.length is 1 and courses[0].roles?.length is 1
```

Comment:
Are you sure? `[][0]` returns `undefined` and the 1st condition of the `if` statement is a guard that `course` is not null.